### PR TITLE
Two simple tasks for building files necessary to support new references.

### PR DESCRIPTION
**Pull request by tfenne**

### CODE REVIEW

**Comment by tfenne** [comment]
@nh13 Could I get a quick thumbs up please?  I think even in full nit-pick mode this is probably a < 2 minute review :smile:

File: tasks/src/main/scala/dagr/tasks/bwa/BwaIndex.scala [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    // Set a sensible default amount of resources required - can still be overridden with requires() later
+    val refSize = ref.toFile.length()
+    if (refSize < 2*1024*1024*1024) {
+      requires(Cores(1), Memory((refSize*5.5).toLong))
+    }
+    else {
+      requires(Cores(1), Memory(refSize*3))
+    }
+  }
+
```
Comment (nh13) [inline tasks/src/main/scala/dagr/tasks/bwa/BwaIndex.scala:45]: you asked for a nitpick, so delete this newline :)

File: tasks/src/main/scala/dagr/tasks/bwa/BwaIndex.scala [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import dagr.tasks.DagrDef.PathToFasta
+
+/**
+ * Indexes a genome for use with BWA.
+ */
+class BwaIndex(val ref: PathToFasta) extends ProcessTask with FixedResources {
+  {
+    // Set a sensible default amount of resources required - can still be overridden with requires() later
+    val refSize = ref.toFile.length()
+    if (refSize < 2*1024*1024*1024) {
```
Comment (nh13) [inline tasks/src/main/scala/dagr/tasks/bwa/BwaIndex.scala]: In my repl, `2*1024*1024*1024` gives `res0: Int = -2147483648`.

**Comment by nh13** [comment]
@tfenne minor concern about overflow, but that's it.

File: tasks/src/main/scala/dagr/tasks/bwa/BwaIndex.scala (hunk 2 above)
Comment (tfenne) [inline tasks/src/main/scala/dagr/tasks/bwa/BwaIndex.scala]: Good catch - thanks.

**Comment by tfenne** [comment]
Updated to address feedback and squashed.  Will merge once everything is green.

**Pull request merged**